src/main/java/io/cdap/e2e/pages/locators/WranglerLocators.java

```diff
@@ -0,0 +1,149 @@
+ * the License.
+ */
+package io.cdap.e2e.pages.locators;
+
+import io.cdap.e2e.utils.SeleniumDriver;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+/**
+ * CDF wrangler related locators.
+ */
+public class WranglerLocators {
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='common-components-scrollableList-scrollDownButton']")
+  public static WebElement scrollButton;
+
+  @FindBy(how = How.XPATH, using = "//div[@class='scroll-down-container text-center disabled']" +
+    "[@data-testid='common-components-scrollableList-scrollDownButton']")
+  public static WebElement scrollButtonDisabled;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='features-dataprep-directives-changeDataType-submenu-decimal" +
+    "-scaleInput']/input")
+  public static WebElement scaleText;
+
+  @FindBy(how = How.XPATH, using = "//button[contains(text(),'Extract')]")
+  public static WebElement extractButton;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-testid='features-dataprep-directives-findAndReplace-applyButton']")
+  public static WebElement replaceAllButton;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-workspace-cli-input']")
+  public static WebElement directiveCommandLine;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-defineVariable" +
+    "-variableNameInput']")
+  public static WebElement variableName;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-testid='features-dataprep-directives-changeDataType-submenu" +
+    "-decimal-applyButton']")
+  public static WebElement applyButtonUppercase;
+
+  @FindBy(how = How.XPATH, using = "//button[contains(text(), 'Apply')]")
+  public static WebElement applyButton;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-defineVariable-" +
+    "conditionTextInput']")
+  public static WebElement variableValue;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-parse-modal-excel-" +
+    "sheetNumberInput']")
+  public static WebElement excelSheetNumber;
+
+  @FindBy(how = How.XPATH, using = "//button[@data-testid='features-dataprep-directives-extractFields-modal-" +
+    "patterns-patternsSelector']")
+  public static WebElement selectPattern;
+
+  @FindBy(how = How.XPATH, using = "//div[contains(@class, 'level-popover')]//input")
+  public static WebElement enterText;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-findAndReplace-oldValueInput']")
+  public static WebElement enterOldValue;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='features-dataprep-directives-encodeDecode-title-Encode']")
+  public static WebElement encodeDirective;
+
+  @FindBy(how = How.XPATH, using = "//div[@data-testid='features-dataprep-directives-encodeDecode-title-Decode']")
```

Comment:
Avoid using the div tag. use //*[@data-testid="..."] pattern instead.

Reply:
done